src/test/test_controller.c

```diff
@@ -1689,6 +1691,187 @@ test_download_status_bridge(void *arg)
+    return mock_microdesc_consensus_cache;
+  }
+}
+
+/** Mock for the function that returns the name of the
+ * file containing the consensuses. */
+char *
+mock_networkstatus_get_cache_fname(int flav,
+                                   const char *flavor_name,
+                                   int unverified_consensus);
+
+char *
+mock_networkstatus_get_cache_fname(int flav,
+                                   const char *flavor_name,
+                                   int unverified_consensus)
+{
+  /** A function needs to do something with
+   * all variables that are passed as arguments */
+  flav = flav & 1;
+  unverified_consensus = unverified_consensus & 1;
+  return tor_strdup(flavor_name);
+}
+
+/** Mock the function that retrieves consensuses
+ *  from a files in the directory. */
+tor_mmap_t *
+mock_tor_mmap_file(const char* filename);
+
+tor_mmap_t *
+mock_tor_mmap_file(const char* filename)
+{
+  tor_mmap_t *res;
+  res = tor_malloc_zero(sizeof(tor_mmap_t));
+  if (!strcmp(filename, "ns")) {
+    res->data = "mock_ns_consensus";
+  }
+  if (!strcmp(filename, "microdesc")) {
+    res->data = "mock_microdesc_consensus";
+  }
+  res->size = strlen(res->data);
+  return res;
+}
+
+int
+mock_tor_munmap_file(tor_mmap_t *handle);
+
+/** Mock the function that clears file data
+ * loaded into the memory */
+int
+mock_tor_munmap_file(tor_mmap_t *handle)
+{
+  tor_free(handle);
+  return 0;
+}
+
+static void
+test_getinfo_helper_dir_from_file(void *arg)
+{
+  /* We just need one of these to pass, it doesn't matter what's in it */
+  control_connection_t dummy;
+  /* Get results out */
+  char *answer = NULL;
+  const char *errmsg = NULL;
+
+  (void)arg;
+
+  setup_bridge_mocks();
+  MOCK(tor_mmap_file, mock_tor_mmap_file);
+  MOCK(tor_munmap_file, mock_tor_munmap_file);
+  MOCK(networkstatus_get_cache_fname,
+       mock_networkstatus_get_cache_fname);
+
+  getinfo_helper_dir(&dummy,
+                     "dir/status-vote/current/consensus",
+                     &answer,
+                     &errmsg);
+  tt_str_op(answer, OP_EQ, "mock_ns_consensus");
+  tt_ptr_op(errmsg, OP_EQ, NULL);
+  tor_free(answer);
+  errmsg = NULL;
+
+  getinfo_helper_dir(&dummy,
+                     "dir/status-vote/current/consensus-microdesc",
+                     &answer,
+                     &errmsg);
+  tt_str_op(answer, OP_EQ, "mock_microdesc_consensus");
+  tt_ptr_op(errmsg, OP_EQ, NULL);
+  errmsg = NULL;
+
+ done:
+  clear_bridge_mocks();
+  tor_free(answer);
+  UNMOCK(tor_mmap_file);
+  UNMOCK(tor_munmap_file);
+  UNMOCK(networkstatus_get_cache_fname);
+  return;
+}
+
+static void
+test_getinfo_helper_dir_from_cache(void *arg)
+{
+  /* We just need one of these to pass, it doesn't matter what's in it */
+  control_connection_t dummy;
+  /* Get results out */
+  char *answer = NULL;
+  const char *errmsg = NULL;
+
+  (void)arg;
+
+  setup_bridge_mocks();
+  MOCK(we_want_to_fetch_flavor, mock_we_want_to_fetch_flavor);
```

Comment:
Why do you mock `we_want_to_fetch_flavor()`? This seems like an interesting function that we should know how to navigate, instead of black-boxing it. Please try to navigate it by using `or_options_t` as suggested above.